examples/extension/README.md

```diff
@@ -1,84 +1,64 @@
 
+The contents in this folder demonstrate how to create an extension for the OpenTelemetry Java instrumentation agent, with examples for every extension point. 
+
+> Read both the source code and the Gradle build script, as they contain documentation that explains the purpose of all the major components.
+
+## Build and execute extension files
+
+To build this extension project, run `./gradlew build`. You can find the resulting jar file in `build/libs/`. 
+
+To execute the extension:
+
+1. Copy the jar file to a host that is running an application you've instrumented using the OpenTelemetry Java instrumentation agent.
+2. Modify the runtime command to add the full path to the extension file. For example:
+
+     ```bash
+     java -javaagent:path/to/opentelemetry-javaagent-all.jar \
+          -Dotel.javaagent.experimental.extensions=build/libs/opentelemetry-java-instrumentation-extension-demo-1.0-all.jar
+          -jar myapp.jar
+     ```
+## Embed extensions in the OpenTelemetry Agent
+
+To simplify deployment, you can embed extensions files into the OpenTelemetry Java Agent to produce a single jar file. With an integrated extension, you no longer need the `-Dotel.javaagent.experimental.extensions` command line option.
+
+For more information, see the `extendedAgent` task in [build.gradle](build.gradle).
 ## Extensions examples
 
-* [DemoIdGenerator](src/main/java/com/example/javaagent/DemoIdGenerator.java) - custom `IdGenerator`
-* [DemoPropagator](src/main/java/com/example/javaagent/DemoPropagator.java) - custom `TextMapPropagator`
-* [DemoPropertySource](src/main/java/com/example/javaagent/DemoPropertySource.java) - default configuration
-* [DemoSampler](src/main/java/com/example/javaagent/DemoSampler.java) - custom `Sampler`
-* [DemoSpanProcessor](src/main/java/com/example/javaagent/DemoSpanProcessor.java) - custom `SpanProcessor`
-* [DemoSpanExporter](src/main/java/com/example/javaagent/DemoSpanExporter.java) - custom `SpanExporter`
-* [DemoServlet3InstrumentationModule](src/main/java/com/example/javaagent/instrumentation/DemoServlet3InstrumentationModule.java) - additional instrumentation
+* Custom `IdGenerator`: [DemoIdGenerator](src/main/java/com/example/javaagent/DemoIdGenerator.java)
+* Custom `TextMapPropagator`: [DemoPropagator](src/main/java/com/example/javaagent/DemoPropagator.java)
+* New default configuration: [DemoPropertySource](src/main/java/com/example/javaagent/DemoPropertySource.java)
+* Custom `Sampler`: [DemoSampler](src/main/java/com/example/javaagent/DemoSampler.java)
+* Custom `SpanProcessor`: [DemoSpanProcessor](src/main/java/com/example/javaagent/DemoSpanProcessor.java)
+* Custom `SpanExporter`: [DemoSpanExporter](src/main/java/com/example/javaagent/DemoSpanExporter.java)
+* Additional instrumentation: [DemoServlet3InstrumentationModule](src/main/java/com/example/javaagent/instrumentation/DemoServlet3InstrumentationModule.java)
+
+## Use cases for extensions
```

Comment:
I think we have to somehow communicate that these examples/use-cases below are not an exhaustive list. They do not cover all of the extensions points that we support and that this module demonstrates.

Reply:
I'll rewrite the headline and intro to clarify that.